src/Container/InMemoryEventStoreFactory.php

```diff
@@ -12,149 +12,26 @@
 
-final class InMemoryEventStoreFactory implements
-    ProvidesDefaultOptions,
-    RequiresConfig,
-    RequiresConfigId
+final class InMemoryEventStoreFactory extends AbstractInMemoryEventStoreFactory
 {
-    use ConfigurationTrait;
-
-    /**
-     * @var string
-     */
-    private $configId;
-
-    /**
-     * Creates a new instance from a specified config, specifically meant to be used as static factory.
-     *
-     * In case you want to use another config key than provided by the factories, you can add the following factory to
-     * your config:
-     *
-     * <code>
-     * <?php
-     * return [
-     *     InMemoryEventStore::class => [InMemoryEventStoreFactory::class, 'service_name'],
-     * ];
-     * </code>
-     *
-     * @throws InvalidArgumentException
-     */
-    public static function __callStatic(string $name, array $arguments): ReadOnlyEventStore
+    protected function createEventStoreInstance(): EventStore
     {
-        if (! isset($arguments[0]) || ! $arguments[0] instanceof ContainerInterface) {
-            throw new InvalidArgumentException(
-                sprintf('The first argument must be of type %s', ContainerInterface::class)
-            );
-        }
-
-        return (new static($name))->__invoke($arguments[0]);
+        return new InMemoryEventStore();
     }
 
-    public function __construct(string $configId = 'default')
-    {
-        $this->configId = $configId;
-    }
-
-    /**
-     * @throws ConfigurationException
-     */
-    public function __invoke(ContainerInterface $container): ReadOnlyEventStore
-    {
-        $config = $container->get('config');
-        $config = $this->options($config, $this->configId);
-
-        $eventStore = new InMemoryEventStore();
-
-        if ($config['read_only']) {
-            $eventStore = new ReadOnlyEventStoreWrapper($eventStore);
-        }
-
-        if (! $config['wrap_action_event_emitter']) {
-            return $eventStore;
-        }
-
-        if (! isset($config['event_emitter'])) {
-            $eventEmitter = new ProophActionEventEmitter(TransactionalActionEventEmitterEventStore::ALL_EVENTS);
-        } else {
-            $eventEmitter = $container->get($config['event_emitter']);
-        }
-
-        $wrapper = new TransactionalActionEventEmitterEventStore($eventStore, $eventEmitter);
-
-        foreach ($config['plugins'] as $pluginAlias) {
-            $plugin = $container->get($pluginAlias);
-
-            if (! $plugin instanceof Plugin) {
-                throw ConfigurationException::configurationError(sprintf(
-                    'Plugin %s does not implement the Plugin interface',
-                    $pluginAlias
-                ));
-            }
+    protected function createActionEventEmitterDecorator(
+        EventStore $eventStore,
+        ActionEventEmitter $actionEventEmitter
+    ): ActionEventEmitterEventStore {
+        assert($eventStore instanceof TransactionalEventStore);
```

Comment:
don't use assert function, use if not and throw exception

Reply:
What is the reason for that? The intention of this assert is just down-type EventStore because in same class I know for sure TransactionalEventStore will be there. The only way it can fail here is to modify this vendor class which would result in failing test. Maybe adding some description for eventual maintainer of this class, in case he would accidentaly not return TransactionalEventStore in createEventStoreInstance() method would help? 
Adding Exception would result in untestable and uncovered code at the line it would be thrown or maybe it would take some ugly magic to do that.